template/interpolate/aws/secretsmanager/secretsmanager.go

```diff
@@ -83,7 +83,11 @@ func getSecretValue(s *SecretString, spec *SecretSpec) (string, error) {
 		return "", err
 	}
 
-	// If key is not set then return first value stored in secret
+	// If key is not set and secret has multiple keys, return error
+	if spec.Key == "" && len(secretValue) > 1 {
+		return "", errors.New("Secret has multiple values and no key was set")
+	}
+
 	if spec.Key == "" {
 		for _, v := range secretValue {
```

Comment:
Would this look better if this condition follows the first condition pattern? 
```
if spec.Key == "" && len(secretValue) == 1 {
     return secretValue[0], nil
} 
```

Reply:
Hi @sylviamoss 

I like your suggestion but as `secretValue` holds a map I need to some other changes for example, change current `if` by:

```
	// Empty key is only valid if secret has a single value
	if spec.Key == "" && len(secretValue) == 1 {
		for _, v := range secretValue {
			return v, nil
		}
	}
```

Then remove https://github.com/overdrive3000/packer/blob/fix-multikey/template/interpolate/aws/secretsmanager/secretsmanager.go#L91-L95 as is not longer needed, in that case I am not sure how to return an error to notify users that if `secret` has multiple keys they must specify a `key`. What do you suggest?

Reply:
The `if` example you gave is exactly what I suggested. It is just to make things more simple in the existing code, without ranging over `secretValue` to simply return the first element.
 
I was talking about the second if statement, the already existing one, not the one you added to check multiple secret values, this one I think is great already!

Reply:
Since the resulting structure is a map there would be  a little extra work to get things in to a indexable slice, which may add a little of complexity to simplify the if. Seeing as this map will only have one or more keys (secret must have at least one key, but can have multiple) I think what is currently written is gtg.

Reply:
I got confused, I thought at first that this was a slice, it's a map which makes totally sense now what you were saying! 🤦 sorry for the confusion